scripts/zones/RuAun_Gardens/Zone.lua

```diff
@@ -5,10 +5,78 @@ local ID = zones[xi.zone.RUAUN_GARDENS]
 -----------------------------------
 local zoneObject = {}
 
-zoneObject.onInitialize = function(zone)
-    for k, v in pairs(ID.npc.PORTALS) do
-        zone:registerTriggerArea(k, unpack(v['coords']))
+local function handleClosingPortal(player, eventId, npcId)
+    if GetNPCByID(npcId):getAnimation() == xi.anim.OPEN_DOOR then
+        player:startOptionalCutscene(eventId)
+    end
+end
+
+local function handleGreenPortal(player)
+    if player:getCharVar('skyShortcut') == 1 then
```

Comment:
later (not today) we should move this to the unlocks table, like so many other one off forever vars we've yet to deal with.

Reply:
no wait I forgot, you can actually -lose- this one after gaining it, never mind! I forgot it breaks.